jest-shared-setup.ts

```diff
@@ -20,3 +21,13 @@ configure({
     return error
   },
 })
+
+afterEach(() => {
+  /**
+   * Clear all mock call counts between tests.
+   * This does NOT clear mock implementations.
+   * Mock implementations are always cleared between test files.
+   */
+  jest.clearAllMocks()
+  resetAllWhenMocks()
```

Comment:
This was previously in `infinite-corridor`'s `setupTests` file, but we should have it everywhere, so I moved it to the shared setup.